tensorflow_lite_support/python/task/processor/proto/search_options_pb2.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Search options protobuf."""
+
+import dataclasses
+from typing import Any, Optional
+
+from tensorflow_lite_support.cc.task.core.proto import external_file_pb2
+from tensorflow_lite_support.cc.task.processor.proto import search_options_pb2
+from tensorflow_lite_support.python.task.core.optional_dependencies import doc_controls
+
+_ExternalFileProto = external_file_pb2.ExternalFile
+_SearchOptionsProto = search_options_pb2.SearchOptions
+
+
+@dataclasses.dataclass
+class SearchOptions:
+  """Options for search processor.
+
+  The index file to search into. Mandatory only if the index is not attached
+  to the output tensor metadata as an AssociatedFile with type SCANN_INDEX_FILE.
+  The index file can be specified by one of the following two ways:
+
+  (1) file contents loaded in `index_file_content`.
+  (2) file path in `index_file_name`.
+
+  If more than one field of these fields is provided, they are used in this
+  precedence order.
+
+  Attributes:
+    index_file_name: Path to the index.
+    index_file_content: The index file contents as bytes.
+    max_results: Maximum number of nearest neighbor results to return.
+  """
+
+  index_file_name: Optional[str] = None
+  index_file_content: Optional[bytes] = None
+  max_results: Optional[int] = 5
+
+  @doc_controls.do_not_generate_docs
+  def to_pb2(self) -> _SearchOptionsProto:
+    """Generates a protobuf object to pass to the C++ layer."""
+    if self.index_file_name is not None or self.index_file_content is not None:
+      return _SearchOptionsProto(
+        index_file=_ExternalFileProto(
+          file_name=self.index_file_name,
+          file_content=self.index_file_content),
+        max_results=self.max_results)
+
```

Comment:
raise a ValueError here if neither index_file_name or index_file_content is specified.

Reply:
@khanhlvg The C++ API seems to handle it already [here](https://github.com/tensorflow/tflite-support/blob/a1d17bf66ef1e148ac6bca2ec3bc81d9993d8421/tensorflow_lite_support/cc/task/core/external_file_handler.cc#L78) though and if both of them are unspecified, it just falls back to using just the `max_results` option [here](https://github.com/tensorflow/tflite-support/pull/823/files#diff-a5fd5f315b938ca84318b6a8b6bd2c2b9cf1e94861512af4699f18832ee6bf08R61).

Reply:
Then you should remove `if self.index_file_name is not None or self.index_file_content is not None:` because Python type check will complain that this function may return None, which doesn't match with the function's return type (_SearchOptionsProto)

Reply:
@khanhlvg But I think a check to switch between the two modes (Embedder/Searcher) is necessary here. The Searcher model doesn't require an index file as illustrated [here](https://github.com/tensorflow/tflite-support/blob/a1d17bf66ef1e148ac6bca2ec3bc81d9993d8421/tensorflow_lite_support/python/test/task/vision/image_searcher_test.py#L105). Do you have any suggestions on how this needs to be handled?

Reply:
Yes there's an option to use the index file bundled in the TFLite file so the `if self.index_file_name is not None or self.index_file_content is not None:` check shouldn't be necessary. I'll remove it when importing this PR for internal review.

Reply:
Ah now I got what you mean. Yes having the if check as you're doing now is correct.